kani-driver/src/cbmc_output_parser.rs

```diff
@@ -0,0 +1,974 @@
+                ("count trailing zeros is undefined for value zero", None),
+                ("count leading zeros is undefined for value zero", None),
+            ],
+        );
+        map.insert("memory-leak", vec![("dynamically allocated memory never freed", None)]);
+        // These pre-conditions should not print temporary variables since they are embedded in the libc implementation.
+        // They are added via `__CPROVER_precondition`.
+        // map.insert("precondition_instance": vec![]);
+        map
+    };
+}
+
+const UNSUPPORTED_CONSTRUCT_DESC: &str = "is not currently supported by Kani";
+const UNWINDING_ASSERT_DESC: &str = "unwinding assertion loop";
+const ASSERTION_FALSE: &str = "assertion false";
+const DEFAULT_ASSERTION: &str = "assertion";
+const REACH_CHECK_DESC: &str = "[KANI_REACHABILITY_CHECK]";
+
+/// Enum that represents a parser item.
+/// See the parser for more information on how they are processed.
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+enum ParserItem {
+    Program {
+        program: String,
+    },
+    #[serde(rename_all = "camelCase")]
+    Message {
+        message_text: String,
+        message_type: String,
+    },
+    Result {
+        result: Vec<Property>,
+    },
+    #[serde(rename_all = "camelCase")]
+    ProverStatus {
+        _c_prover_status: String,
+    },
+}
+
+impl ParserItem {
+    /// Determines if an item must be skipped or not.
+    fn must_be_skipped(&self) -> bool {
+        matches!(&self, ParserItem::Message { message_text, .. } if message_text.starts_with("Building error trace"))
+            || matches!(&self, ParserItem::Message { message_text, .. } if message_text.starts_with("VERIFICATION"))
+    }
+}
+
+/// Struct that represents a property.
+///
+/// Note: `reach` is not part of the parsed data, but it's useful to annotate
+/// its reachability status.
+#[derive(Clone, Debug, Deserialize)]
+pub struct Property {
+    pub description: String,
+    pub property: String,
+    #[serde(rename = "sourceLocation")]
+    pub source_location: SourceLocation,
+    pub status: CheckStatus,
+    pub reach: Option<CheckStatus>,
+    pub trace: Option<Vec<TraceItem>>,
+}
+
+/// Struct that represents a source location.
+///
+/// Source locations may be completely empty, which is why
+/// all members are optional.
+#[derive(Clone, Debug, Deserialize)]
+pub struct SourceLocation {
+    pub column: Option<String>,
+    pub file: Option<String>,
+    pub function: Option<String>,
+    pub line: Option<String>,
+}
+
+impl SourceLocation {
+    /// Determines if fundamental parts of a source location are missing.
+    fn is_missing(&self) -> bool {
+        self.file.is_none() && self.function.is_none()
+    }
+}
+
+/// `Display` implement for `SourceLocation`.
+///
+/// This is used to format source locations for individual checks. But source
+/// locations may be printed in a different way in other places (e.g., in the
+/// "Failed Checks" summary at the end).
```

Comment:
Can you describe what the formatted location looks like? It's hard to infer from reading the code.

Reply:
Added a description, let me know how it looks!